packages/material-ui-system/src/styleFunctionSx.js

```diff
@@ -40,4 +47,18 @@ function css(styleFunction) {
   return newStyleFunction;
 }
 
-export default css;
+/**
+ *
+ * @deprecated
+ * The css style function is deprecated. Use the `styleFunctionSx` instead.
+ */
+export function css(styleFunction) {
+  if (process.env.NODE_ENV !== 'production') {
+    console.warn(
+      'Material-UI: The `css` function is deprecated. Use the `styleFunctionSx` instead.',
+    );
+  }
```

Comment:
Question: Can we tell the system that certain system-functions apply arbitrary `.propTypes`? We've learned in the Typography v2 migration that console warnings aren't all that helpful for issues during render.

Reply:
I don't understand the question I think. All system function apply the proptypes that they expect the component to have, depending on which style function is used. 

Btw this warnings is just for the sake of using deprecated method from the system. Here is an example:

```diff
-import { css } from '@material-ui/system';
+import { styleFunctionSx } from '@material-ui/system';
import styled from 'styled-components';

-const CustomComponent = styled('div')(css);
+const CustomComponent = styled('div')(styledFunctionSx);

```

I've moved the props validation in the `propTypes`.

Reply:
Could you point me to the test that is covering this warning at the moment?

Reply:
Sure, here it is https://github.com/mui-org/material-ui/pull/23480/files#diff-0e470568cc7fbe1ce5a72667f25efeb6f61eabee1c423dc677369da32dfcc8b0R38-R45